### PR TITLE
Implement TranscriptStream actor with pub/sub dispatch (BT-371)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Implements `beamtalk_transcript_stream.erl` as a gen_server actor — a shared workspace log with pub/sub semantics, as specified in ADR 0010 (Phase 1).

**Linear issue:** https://linear.app/beamtalk/issue/BT-371

## Key Changes

- **`beamtalk_transcript_stream.erl`** — New gen_server with:
  - `show:` / `cr` (async cast) — buffer text + push to subscribers
  - `subscribe` / `unsubscribe` (async cast) — with process monitor auto-cleanup
  - `recent` / `clear` (sync call) — buffer inspection
  - Ring buffer (`queue:queue()`) with configurable max size (default 1000)
  - `has_method/1` for dispatch compatibility
  - Input validation: non-pid subscribe silently ignored, invalid max_buffer rejected

- **`beamtalk_transcript_stream_tests.erl`** — 19 unit tests covering:
  - Buffer operations (show, cr, recent, clear, overflow)
  - Subscriber lifecycle (subscribe, unsubscribe, dead subscriber auto-removal)
  - Edge cases (duplicate subscribe, non-pid subscribe, invalid max_buffer)
  - Spawn API (linked and unlinked)

## Design Decisions

- Coexists with existing `transcript.erl` (no breaking changes)
- Follows ADR 0010: cast for I/O side effects, call for queries
- Uses `#beamtalk_error{}` for unknown method calls
- `to_string/1` matches existing `transcript.erl` conversion logic

### CODE REVIEW

File: runtime/apps/beamtalk_runtime/test/beamtalk_transcript_stream_tests.erl [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    timer:sleep(10),
+    Result = gen_server:call(Pid, recent),
+    ?assertEqual([<<"42">>], Result),
+    gen_server:stop(Pid).
+
+show_converts_atom_test() ->
+    {ok, Pid} = beamtalk_transcript_stream:start_link(),
+    gen_server:cast(Pid, {'show:', hello}),
+    timer:sleep(10),
+    Result = gen_server:call(Pid, recent),
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/test/beamtalk_transcript_stream_tests.erl]: Intentionally not changing to  format. This is Phase 1 per ADR 0010 — the gen_server is tested with its own protocol. Dispatch protocol integration (including Future resolution) is Phase 3 scope when codegen is updated to route `Transcript show:` through this actor.

File: runtime/apps/beamtalk_runtime/src/beamtalk_transcript_stream.erl [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    {noreply, State1};
+handle_cast(cr, State) ->
+    Text = <<"\n">>,
+    State1 = buffer_text(Text, State),
+    push_to_subscribers(Text, State1),
+    {noreply, State1};
+handle_cast({subscribe, Pid}, State) ->
+    {noreply, add_subscriber(Pid, State)};
+handle_cast({unsubscribe, Pid}, State) ->
+    {noreply, remove_subscriber(Pid, State)};
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/src/beamtalk_transcript_stream.erl]: Out of scope for Phase 1. The acceptance criteria explicitly exclude codegen changes (Phase 3). The current `handle_cast` clauses use a simple protocol for direct Erlang usage. When Phase 3 integrates this with Beamtalk dispatch, we'll add `{Selector, Args, FuturePid}` handlers that resolve/reject futures.

File: runtime/apps/beamtalk_runtime/src/beamtalk_transcript_stream.erl [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+%% @private
+-spec handle_call(term(), {pid(), term()}, state()) ->
+    {reply, term(), state()}.
+handle_call(recent, _From, #state{buffer = Buffer} = State) ->
+    {reply, queue:to_list(Buffer), State};
+handle_call(clear, _From, State) ->
+    {reply, ok, State#state{buffer = queue:new(), buffer_size = 0}};
+handle_call(_Request, _From, State) ->
+    Error0 = beamtalk_error:new(does_not_understand, 'TranscriptStream'),
+    {reply, {error, Error0}, State}.
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/src/beamtalk_transcript_stream.erl]: Fixed in 1e289af. Now uses `beamtalk_error:with_selector/2` on the unknown-request DNU error, consistent with `beamtalk_actor:handle_dnu/4`.
